Pod/Classes/OAStackView.m

```diff
@@ -141,6 +141,16 @@ - (void)setAlignmentConstraints {
   [self.alignmentStrategy alignLastView:self.subviews.lastObject];
 }
 
+- (void)setAlignmentStrategy:(OAStackViewAlignmentStrategy *)alignmentStrategy {
+  [_alignmentStrategy removeAddedConstraints];
```

Comment:
Of course, sorry for confusion:

every objective-c setter looks like following under the hood:
```
- (void)setFoo(Foo *)foo {
   if (_foo isEqual:foo) {
     return; // there is no point to perform side effects, value of property has not changed
   }
   _foo = foo;
   //side effects
}
```